machine-learning/Dockerfile

```diff
@@ -17,6 +17,41 @@ RUN mkdir /opt/armnn && \
     cd /opt/ann && \
     sh build.sh
 
+# Warning: 57.2Gb of disk space required to pull this image
+# https://github.com/microsoft/onnxruntime/blob/main/dockerfiles/Dockerfile.rocm
+FROM rocm/pytorch:rocm6.1.2_ubuntu22.04_py3.10_pytorch_release-2.1.2 as builder-whl-rocm
```

Comment:
We shouldn't need PyTorch at all for this. Does `rocm/dev-ubuntu-22.04:6.1.2-complete` work?

Reply:
This'll definitely need smaller upstream images, the actions runners simply do not have 60 gigs of disk space to pull it.

Reply:
I think the image I mentioned is small enough: 4gb compressed vs 17gb for the one with PyTorch. But I'm not sure if there was a reason the ONNX Runtime folks used the PyTorch variant for their Dockerfile.

Reply:
I've tested with `rocm/dev-ubuntu-22.04:6.1.2-complete`, here are my conlusions:
 * Extracted image is 26.3GB
 * Compilation of Onnxruntime is okay
 * Final image 27.7GB (compared to 32.9GB)
 * Works like a charm, even in parallel (tested with 3 **Smart Search**, 4 **Face detection** and 1 **Facial Recognition**)

I still have to get rid of `libmimalloc`, otherwise it crashes.